Sources/SwiftParser/Modifiers.swift

```diff
@@ -18,36 +18,38 @@ extension Parser {
     var elements = [RawDeclModifierSyntax]()
     var modifierLoopCondition = LoopProgressCondition()
     MODIFIER_LOOP: while modifierLoopCondition.evaluate(currentToken) {
-      switch self.at(anyIn: DeclarationModifier.self) {
-      case (.private, _)?,
-        (.fileprivate, _)?,
-        (.internal, _)?,
-        (.public, _)?:
+      switch self.canRecoverTo(anyIn: DeclarationStart.self, overrideRecoveryPrecedence: .declKeyword) {
```

Comment:
I think this should be the easiest way. Since we get the `spec` from the associated value in `declarationModifier` and `declarationKeyword`, override the token precedence in `DeclarationStart` may require us to add a new initializer to `TokenSpec` which init from other `TokenSpec` and pass the new token precedence to it. Is there anything I missed?

Reply:
The thing is that every other call that tries to recover to a `DeclarationModifier` or `DeclarationStart` would also need to specify the `overrideRecoveryPrecedence`. And really, the increased recovery precedence is not because of the context we are recovering here but it’s inherent to the keywords we try to recover to, all of which are declaration keywords in this case. 

I briefly checked locally and adjusting the recovery precedence of the `DeclarationModifier`s seems to work. So, I think we should continue doing that.

Reply:
> the increased recovery precedence is not because of the context we are recovering here

That makes sense👍. I've moved the precedence override to `DeclarationModifier`.